homeassistant/components/sensor/linky.py

```diff
@@ -0,0 +1,84 @@
+"""
+Support for Linky.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor.linky/
+"""
+import logging
+import json
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pylinky==0.1.5']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'linky'
+
+SCAN_INTERVAL = timedelta(minutes=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Configure the platform and add the Linky sensor."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    from pylinky.client import LinkyClient
+    client = LinkyClient(username, password)
```

Comment:
Please check if the credentials are valid. If not, make the platform setup fail with a log entry. Otherwise will the users end-up with a platform which is not working in their instance.

Reply:
Actually, the constructor is doing nothing except storing credentials.
`fetch_data` is the only method that login the user and load its data, that's the reason why it's inside the try/catch block.

Reply:
You should make the first call in `setup_platform` in this case to ensure that the credentials are valid.